SpatialGDK/Source/SpatialGDK/Private/Interop/WellKnownEntitySystem.cpp

```diff
@@ -103,26 +179,56 @@ void WellKnownEntitySystem::ProcessAuthorityGain(const Worker_EntityId EntityId,
 {
 	GlobalStateManager->AuthorityChanged({ EntityId, ComponentSetId, WORKER_AUTHORITY_AUTHORITATIVE });
 
-	if (SubView->GetView()[EntityId].Components.ContainsByPredicate(
+	if (GdkSubView->GetView()[EntityId].Components.ContainsByPredicate(
 			SpatialGDK::ComponentIdEquality{ SpatialConstants::SERVER_WORKER_COMPONENT_ID }))
 	{
 		GlobalStateManager->TrySendWorkerReadyToBeginPlay();
 	}
 
-	if (SubView->GetView()[EntityId].Components.ContainsByPredicate(
+	if (GdkSubView->GetView()[EntityId].Components.ContainsByPredicate(
 			SpatialGDK::ComponentIdEquality{ SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID }))
 	{
 		InitializeVirtualWorkerTranslationManager();
+		VirtualWorkerTranslationManager->SetKnownServerSystemEntities(GetAllServerSystemEntities());
 		VirtualWorkerTranslationManager->AuthorityChanged({ EntityId, ComponentSetId, WORKER_AUTHORITY_AUTHORITATIVE });
 	}
 }
 
+TArray<Worker_EntityId> WellKnownEntitySystem::GetAllServerSystemEntities() const
+{
+	TArray<Worker_EntityId> ServerSystemEntities;
+
+	for (const auto& Iter : SystemEntitySubView->GetView())
+	{
+		const Worker_EntityId EntityId = Iter.Key;
+		const SpatialGDK::EntityViewElement& Element = Iter.Value;
+
+		// Get all worker system entities.
+		const ComponentData* ServerSystemWorkerData = Element.Components.FindByPredicate([](const auto& CompData) {
+			return CompData.GetComponentId() == SpatialConstants::WORKER_COMPONENT_ID;
+		});
+
+		if (ServerSystemWorkerData == nullptr)
+		{
+			continue;
```

Comment:
Is this an error?

Reply:
That depends if Runtime create any system entities that aren't also worker entities. If not now, could be in future?

Reply:
Yep I meant, should we add an error log here